model-server/src/main/kotlin/org/modelix/model/server/handlers/KeyValueLikeModelServer.kt

```diff
@@ -359,24 +349,46 @@ class KeyValueLikeModelServer(
 
     @Throws(IOException::class)
     private fun CallContext.checkKeyPermission(key: String, type: EPermissionType) {
-        if (key.startsWith(PROTECTED_PREFIX)) {
-            throw NoPermissionException("Access to keys starting with '$PROTECTED_PREFIX' is only permitted to the model server itself.")
-        }
-        if (key.startsWith(RepositoriesManager.KEY_PREFIX)) {
-            throw NoPermissionException("Access to keys starting with '${RepositoriesManager.KEY_PREFIX}' is only permitted to the model server itself.")
-        }
-        if ((key == RepositoriesManager.LEGACY_SERVER_ID_KEY || key == RepositoriesManager.LEGACY_SERVER_ID_KEY2) && type.includes(
-                EPermissionType.WRITE,
-            )
-        ) {
-            throw NoPermissionException("'$key' is read-only.")
-        }
-        if (HashUtil.isSha256(key)) {
-            // Reading entries with a hash key is equivalent to uncompressing data that the user already has access to.
-            // If he isn't allowed to read the entry then he shouldn't be allowed to know the hash.
-            // A permission check has happened somewhere earlier.
-            return
+        val isWrite = type == EPermissionType.WRITE
+        switchKeyType(
+            key = key,
+            immutableObject = {
+                // This legacy API only supports repositories in non-isolated storage.
```

Comment:
Should we introduce `legacy-global-objects/write` and checking `legacy-global-objects/read`/`legacy-global-objects/write`?

Reply:
Added those permissions. They are granted to anyone who has access to at least one repository.